lock/src/main/java/com/auth0/android/lock/LockActivity.java

```diff
@@ -342,6 +351,43 @@ public void onDatabaseAuthenticationRequest(DatabaseChangePasswordEvent event) {
         request.start(changePwdCallback);
     }
 
+    @SuppressWarnings("unused")
+    @Subscribe
+    public void onEnterpriseAuthenticationRequest(EnterpriseWebLoginEvent event) {
+        //same as social authentication
+        if (options == null) {
+            return;
+        }
+
+        progress.showProgress();
+        String pkgName = getApplicationContext().getPackageName();
+        CallbackHelper helper = new CallbackHelper(pkgName);
+        lastIdp = new WebIdentityProvider(helper, options.getAccount(), idpCallback);
+        lastIdp.setUseBrowser(options.useBrowser());
+        lastIdp.setParameters(options.getAuthenticationParameters());
+        lastIdp.start(LockActivity.this, event.getConnectionName());
+    }
+
+    @SuppressWarnings("unused")
+    @Subscribe
+    public void onEnterpriseAuthenticationRequest(EnterpriseROLoginEvent event) {
```

Comment:
I'm not buying separating the enterprise login with 2 different event. I'd rather make the decision in a single method (by fetching the ro enabled flag from the event), also it will make things easier when we add an option to go always with web flow.